src/Microsoft.AspNetCore.Sockets.Http/Internal/Transports/LongPollingTransport.cs

```diff
@@ -16,18 +17,25 @@ public class LongPollingTransport : IHttpTransport
         private readonly ChannelReader<byte[]> _application;
         private readonly ILogger _logger;
         private readonly CancellationToken _timeoutToken;
+        private readonly TimeSpan _timeout;
         private readonly string _connectionId;
 
-        public LongPollingTransport(CancellationToken timeoutToken, ChannelReader<byte[]> application, string connectionId, ILoggerFactory loggerFactory)
+        public LongPollingTransport(CancellationToken timeoutToken, TimeSpan timeout, ChannelReader<byte[]> application, string connectionId, ILoggerFactory loggerFactory)
         {
             _timeoutToken = timeoutToken;
+            _timeout = timeout;
             _application = application;
             _connectionId = connectionId;
             _logger = loggerFactory.CreateLogger<LongPollingTransport>();
         }
 
-        public async Task ProcessRequestAsync(HttpContext context, CancellationToken token)
+        public async Task ProcessRequestAsync(ConnectionContext connection, HttpContext context, CancellationToken token)
         {
+            if (connection.Features.Get<IConnectionInherentKeepAliveFeature>() == null)
```

Comment:
This still makes me cringe. Do we call this before we call into the end point?

Reply:
Hrm. No we don't. Unfortunately I don't see an easy way to arrange this such that we do. It works today because of Negotiate and the way our client works, since the client always establishes the poll before sending the negotiate frame. As a result, `OnConnectionAsync` is still waiting for the negotiate message by the time the first poll comes in a sets the feature.

This could be a tricky one to figure out. I'm not sure the best way to rearrange this to make it work, but I'll take a look. Short of requiring the client to start the poll/SSE transport before even calling OnConnectedAsync, I'm not sure what to do.

Reply:
What if the Channels along the path didn't "activate" until the transport was initialized? So `WaitForReadAsync` blocked until the first poll/event stream request came in. Then we check the feature after the first WaitForReadAsync?